### ISSUE TITLE
Endless loop on request to memcache

### ISSUE THREAD

**Issue by wreckah** [gabrielfalcao/HTTPretty#184]
I've got an endless loop on `flush_all` request from Django caching backend to memcached with `httpretty.enabled`:
```py
File "manage.py", line 25, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/test.py", line 50, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/test.py", line 71, in execute
    super(Command, self).execute(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/test.py", line 88, in handle
    failures = test_runner.run_tests(test_labels)
  File "/usr/local/lib/python2.7/dist-packages/django/test/runner.py", line 146, in run_tests
    result = self.run_suite(suite)
  File "/usr/local/lib/python2.7/dist-packages/django/test/runner.py", line 113, in run_suite
    ).run(suite)
  File "/usr/lib/python2.7/unittest/runner.py", line 151, in run
    test(result)
  File "/usr/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/usr/local/lib/python2.7/dist-packages/django/test/testcases.py", line 184, in __call__
    super(SimpleTestCase, self).__call__(result)
  File "/usr/lib/python2.7/unittest/case.py", line 395, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/case.py", line 367, in run
    cleanUpSuccess = self.doCleanups()
  File "/usr/lib/python2.7/unittest/case.py", line 386, in doCleanups
    function(*args, **kwargs)
  File "/home/wreckah/projects/target-web/web/test_utils/testcases.py", line 23, in _cache_cleanup
    get_cache('default').clear()

----------------------------------------------
  File "/usr/local/lib/python2.7/dist-packages/django/core/cache/backends/memcached.py", line 150, in clear
    self._cache.flush_all()
  File "/usr/local/lib/python2.7/dist-packages/memcache.py", line 321, in flush_all
    s.flush()
  File "/usr/local/lib/python2.7/dist-packages/memcache.py", line 1215, in flush
    self.send_cmd('flush_all')
  File "/usr/local/lib/python2.7/dist-packages/memcache.py", line 1161, in send_cmd
    self.socket.sendall(cmd + '\r\n')
  File "/usr/local/lib/python2.7/dist-packages/httpretty/core.py", line 361, in sendall
    return self.real_sendall(data, *args, **kw)
  File "/usr/local/lib/python2.7/dist-packages/httpretty/core.py", line 338, in real_sendall
    received = self.truesock.recv(self._bufsize)**
----------------------------------------------

  File "/usr/lib/python2.7/unittest/signals.py", line 36, in __call__
    self.default_handler(signum, frame)
KeyboardInterrupt
```
Seems that it's occurs because of https://github.com/gabrielfalcao/HTTPretty/commit/44d6484fe35ae776e32b04526580659afc86c72b
```py
        def real_sendall(self, data, *args, **kw):
            if self.is_http:  # no need to connect if `self.is_http` is
                self.truesock.connect(self._address)

            self.truesock.settimeout(0)
            self.truesock.sendall(data, *args, **kw)

            should_continue = True
            while should_continue:
                try:
                    received = self.truesock.recv(self._bufsize)
                    self.fd.write(received)
                    should_continue = len(received) > 0

                except socket.error as e:
                    if e.errno == EAGAIN:
                        continue
                    break

            self.fd.seek(0)
```
On the first iteration of `while should_continue` `received` has `OK\r\n` value. The second iteration raises `socket.error` with `errno = EAGAIN`. Then because of `should_continue = True` loop will never exit.

I've tried to find out what `EAGAIN` means: http://stackoverflow.com/questions/4058368/what-does-eagain-mean. Looks like memcache returns it as the end of response.

Unfortunately I cannot solve problem by myself. So there is no merge request.
  
Environment:
  * CPython 2.7.6
  * httpretty 0.8.3
  * memcached 1.4.14
  * python-memcached 1.53
  * Django 1.6.2

**Comment by mpaolini** [comment]
httpretty needs to be quite aggressive on the underlying socket to intercept http(s) traffic.

in #208 I bypass all the intercepting logic for non-http(s) sockets. Works for me now.

**Comment by gabrielfalcao** [comment]
This might have been fixed by @frankamp on #216
